apps/api/src/app/inbox/utils/notification-mapper.ts

```diff
@@ -0,0 +1,69 @@
+import type { MessageEntity } from '@novu/dal';
+import { ButtonTypeEnum } from '@novu/shared';
+
+import type { InboxNotification, Subscriber } from './types';
+
+const mapSingleItem = ({
+  _id,
+  content,
+  read,
+  archived,
+  createdAt,
+  lastReadDate,
+  archivedAt,
+  channel,
+  subscriber,
+  actorSubscriber,
+  actor,
+  cta,
+  tags,
+}: MessageEntity): InboxNotification => {
+  const to: Subscriber = {
+    id: subscriber?._id ?? '',
+    firstName: subscriber?.firstName,
+    lastName: subscriber?.lastName,
+    avatar: subscriber?.avatar,
+    subscriberId: subscriber?.subscriberId ?? '',
+  };
+  const primaryCta = cta.action?.buttons?.find((button) => button.type === ButtonTypeEnum.PRIMARY);
+  const secondaryCta = cta.action?.buttons?.find((button) => button.type === ButtonTypeEnum.SECONDARY);
+
+  return {
+    id: _id,
+    // subject: subject,
+    body: content as string,
+    to,
+    read,
+    archived,
+    createdAt,
+    readAt: lastReadDate,
+    archivedAt,
+    actor: actorSubscriber
```

Comment:
PTAL at [this](https://www.notion.so/Notifications-DSL-dc7b2a8c7c6b49bf95e770e7c94d9faa?d=8e22b8a722c44f348a40d759a6df3d21&pvs=4#42c8642c1b2b4688979a98a8396db5e0).